### PR TITLE
Fixed Parameterized testing

### LINKED ISSUE #1183: ROS2 branch is failing
**Description**

The `ros2` branch is failing on ros2 master builds. See output below, looks like its related to gtest and depreciated feature use. CI job in Nav2 validates: https://app.circleci.com/pipelines/github/ros-planning/navigation2/4053/workflows/2a54ca7b-1b38-4182-b45e-b4a638d20760/jobs/15522

**To Reproduce**

Build `ros2` based on `ros2 master` 

**Expected behavior**

Builds correctly :-)
**Environment (please complete the following information):**

 - OS: Ubuntu 20
 - Gazebo version: N/A
 - ROS version: ROS2 master
  - gazebo_ros_pkgs: Source of ros2 branch
 - glx server in use: N/A

**Additional context**

```
]0;colcon build [18/21 done] [0 ongoing]Starting >>> gazebo_ros
--- stderr: gazebo_ros
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:38: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:38: note: in definition of macro ‘INSTANTIATE_TEST_CASE_P’
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:73: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                                                         ^
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:496:73: warning: ‘constexpr bool testing::internal::InstantiateTestCase_P_IsDeprecated()’ is deprecated: INSTANTIATE_TEST_CASE_P is deprecated, please use INSTANTIATE_TEST_SUITE_P [-Wdeprecated-declarations]
  496 |   static_assert(::testing::internal::InstantiateTestCase_P_IsDeprecated(), \
      |                                                                         ^
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:62,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-internal.h:1210:16: note: declared here
 1210 | constexpr bool InstantiateTestCase_P_IsDeprecated() { return true; }
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp: In function ‘std::string gtest_PluginsTestPlugins_EvalGenerateName_(const testing::TestParamInfo<TestParams>&)’:
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:472:43: error: expected primary-expression before ‘)’ token
  472 |       auto t = std::make_tuple(__VA_ARGS__);                                  \
      |                                           ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:473:48: error: template argument 1 is invalid
  473 |       static_assert(std::tuple_size<decltype(t)>::value <= 2,                 \
      |                                                ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:479:23: error: expected primary-expression before ‘)’ token
  479 |         DUMMY_PARAM_))))(info);                                               \
      |                       ^
/opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:498:3: note: in expansion of macro ‘INSTANTIATE_TEST_SUITE_P’
  498 |   INSTANTIATE_TEST_SUITE_P
      |   ^~~~~~~~~~~~~~~~~~~~~~~~
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1: note: in expansion of macro ‘INSTANTIATE_TEST_CASE_P’
   70 | INSTANTIATE_TEST_CASE_P(
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest-param-test.h:181,
                 from /opt/ros/foxy/src/gtest_vendor/include/gtest/gtest.h:67,
                 from /opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:15:
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-param-util.h: In instantiation of ‘void testing::internal::TestNotEmpty() [with T = int]’:
/opt/underlay_ws/src/ros-simulation/gazebo_ros_pkgs/gazebo_ros/test/test_plugins.cpp:70:1:   required from here
/opt/ros/foxy/src/gtest_vendor/include/gtest/internal/gtest-param-util.h:382:27: error: static assertion failed: Empty arguments are not allowed.
  382 |   static_assert(sizeof(T) == 0, "Empty arguments are not allowed.");
      |                 ~~~~~~~~~~^~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-inconsistent-missing-override’
make[2]: *** [test/CMakeFiles/test_plugins.dir/build.make:63: test/CMakeFiles/test_plugins.dir/test_plugins.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:530: test/CMakeFiles/test_plugins.dir/all] Error 2
make: *** [Makefile:141: all] Error 2
---
Failed   <<< gazebo_ros [23.7s, exited with code 2]
]0;colcon build [18/21 done] [1 ongoing]]0;colcon build [19/21 done] [0 ongoing]
Summary: 18 packages finished [1min 44s]
  1 package failed: gazebo_ros
  2 packages had stderr output: behaviortree_cpp_v3 gazebo_ros
  2 packages not processed
```

### PR DESCRIPTION
**Pull request by ahcorde**
Related with this issue https://github.com/ros-simulation/gazebo_ros_pkgs/issues/1183

### CODE REVIEW

**Review by ruffsl — APPROVED** [review]

File: gazebo_ros/test/test_plugins.cpp [hunk 1]
```diff
@@ -80,7 +80,7 @@ INSTANTIATE_TEST_CASE_P(
         "worlds/ros_world_plugin.world"}, {"test"}}),
     TestParams({{"-s", "libgazebo_ros_init.so", "worlds/sdf_node_plugin.world"}, {"/foo/my_topic"}})
     // cppcheck-suppress syntaxError
-  ), );
+  ));
```
Comment (jacobperron) [inline gazebo_ros/test/test_plugins.cpp]: I'm just guessing that this change is what is responsible or the [uncrustify errors](http://build.ros2.org/job/Rpr__gazebo_ros_pkgs__ubuntu_focal_amd64/23/)

File: gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp [hunk 2]
```diff
@@ -122,12 +122,12 @@ TEST_P(GazeboRosDiffDriveTest, Publishing)
...
 }
 
-INSTANTIATE_TEST_CASE_P(
+INSTANTIATE_TEST_SUITE_P(
   GazeboRosDiffDrive, GazeboRosDiffDriveTest, ::testing::Values(
     TestParams({"worlds/gazebo_ros_diff_drive.world"}),
     TestParams({"worlds/gazebo_ros_skid_steer_drive.world"})
     // cppcheck-suppress syntaxError
-  ), );
+  ));
```
Comment (jacobperron) [inline gazebo_plugins/test/test_gazebo_ros_diff_drive.cpp]: here too

**Review by jacobperron — COMMENTED** [review]
The Foxy build is failing, I guess this change isn't compatible with Foxy?

If we can't make it compatible, perhaps it's time to create `foxy` branch.

**Comment by ahcorde** [comment]
we should probably create a `foxy` branch because the `gtest` version has changed in `rolling`

**Comment by ahcorde** [comment]
@jacobperron Who is in charge of creating a `foxy` branch ?

**Comment by scpeters** [comment]
> we should probably create a `foxy` branch because the `gtest` version has changed in `rolling`

should we branch off the head of `ros2` or an earlier commit?

**Comment by SteveMacenski** [comment]
I think if you branch `foxy` off of the current `ros2` we'll be good to rock and roll. Current `ros2` works with `foxy`, after this PR it will work with `ros2 master`.

**Review by SteveMacenski — APPROVED** [review]

**Comment by jacobperron** [comment]
I've created a foxy branch. Related PRs:

- https://github.com/ros/rosdistro/pull/27180
- https://github.com/ros2-gbp/gazebo_ros_pkgs-release/pull/1

**Comment by ruffsl** [comment]
> I've created a foxy branch. Related PRs:

@jacobperron  Looks like the referenced PRs are merged. Is this ready now?

**Comment by SteveMacenski** [comment]
+1 on ready to go

**Comment by jacobperron** [comment]
@ros-pull-request-builder retest this please

**Comment by jacobperron** [comment]
@osrf-jenkins run tests please

**Comment by jacobperron** [comment]
@j-rivero @chapulina I've taken care of the Foxy builds for the ROS buildfarm; now they should only be triggered for PRs against the `foxy` branch. How can we do the same for the OSRF buildfarm?

**Comment by chapulina** [comment]
> How can we do the same for the OSRF buildfarm?

https://github.com/ignition-tooling/release-tools/pull/338

**Review by chapulina — APPROVED** [review]

**Comment by chapulina** [comment]
I believe everything is in place. Merging!

**Pull request merged**